src/compile/scale/properties.ts

```diff
@@ -120,7 +120,8 @@ function parseUnitScaleProperty(model: UnitModel, property: Exclude<keyof (Scale
                 domainMax: specifiedScale.domainMax,
                 markDef,
                 config,
-                hasNestedOffsetScale: channelHasNestedOffsetScale(encoding, channel)
+                hasNestedOffsetScale: channelHasNestedOffsetScale(encoding, channel),
+                hasRangeChannels: encodingHasRangeChannels(encoding)
```

Comment:
The fact that `encodingHasRangeChannels(encoding)` doesn't take in `channel` seems alarming to me. 

Basically, Y2 shouldn't affect x-scale's zero.  

I think you should 
- undo adding `encodingHasRangeChannels`
- rename `hasRangeChannels` to `hasRangeChannel` and just check if there is x2 for x or y2 for x.  (And maybe radius2 for radius.)